streams/src/test/java/org/apache/kafka/streams/integration/TableTableJoinIntegrationTest.java

```diff
@@ -79,7 +79,7 @@ public void prepareTopology() throws InterruptedException {
         @Override
         public void apply(final Long key, final String value) {
             numRecordsExpected++;
-            if (value.equals(expected)) {
+            if (value != null && value.equals(expected)) {
```

Comment:
nit: Can we just call `expected.equals(value)` which does null check internally as well?